web_client/dialogs/editRegionOfInterest.js

```diff
@@ -0,0 +1,207 @@
+            this._magnification = this.areaElement.maxMag;
+        }
+        var bounds = this.scaleBounds();
+        this.$el.html(
+            editRegionOfInterest({
+                magnification: this._magnification,
+                scaleWidth: bounds.width,
+                scaleHeight: bounds.height,
+                maxMag: this.areaElement.maxMag,
+                numberOfPixel: this.getNumberPixels(),
+                fileSize: this.getConvertFileSize()
+            })
+        ).girderModal(this);
+        this.updateform();
+        return this;
+    },
+
+    /**
+     * Convert from zoom level to magnification.
+     */
+    zoomToMagnification(zoom) {
+        return Math.round(parseFloat(this.areaElement.maxMag) *
+            Math.pow(2, zoom - parseFloat(this.areaElement.maxZoom)) * 10) / 10;
+    },
+
+    /**
+     * Convert from magnification to zoom level.
+     */
+    magnificationToZoom(magnification) {
+        return parseFloat(this.areaElement.maxZoom) -
+            Math.log2(this.areaElement.maxMag / magnification);
+    },
+
+    /**
+     * Convert from magnification to zoom level.
+     */
+    scaleBounds() {
+        var zoom = this.magnificationToZoom(this._magnification);
+        var factor = Math.pow(2, zoom - this.areaElement.maxZoom);
+        var scaleWidth = Math.round(factor * this.areaElement.width);
+        var scaleHeight = Math.round(factor * this.areaElement.height);
+        return { 'width': scaleWidth, 'height': scaleHeight };
+    },
+
+    /**
+     * Get the number of pixel in the region of interest
+     */
+    getNumberPixels() {
+        var bounds = this.scaleBounds();
+        var Npixel = bounds.width * bounds.height;
+        return Npixel;
+    },
+
+    /**
+     * Get the size of the file before download it for an image in 24b/px (result in Bytes)
+     */
+    getFileSize() {
+        var fileSize = (this.getNumberPixels() * 3 + this._sizeCte) * this._compressionRatio;
+        return fileSize;
+    },
+
+    /**
+     * Get the size of the file in the appropriate unity (Bytes, MB, GB...)
+     */
+    getConvertFileSize() {
+        var Nbytes = this.getFileSize();
+        var convertedSize = formatSize(Nbytes);
+        if (Nbytes >= Math.pow(2, 30)) {
+            this.downloadDisable(true);
+        } else {
+            this.downloadDisable(false);
+        }
+        return convertedSize;
+    },
+
+    /**
+     * Disable the Download button if SizeFile > 1GB
+     */
+    downloadDisable(bool) {
+        var element = $('#msgDisable').attr('id');
+        if (bool === true) {
+            $('#download-submit').attr('disabled', 'disabled');
+            if (typeof element === typeof undefined) {
+                var msgDisable = $('<span></span>').text('Size > 1GB : Impossible Download ');
+                msgDisable.attr('id', 'msgDisable');
+                msgDisable.css({'color': 'red', 'margin-right': '120px'});
```

Comment:
Move custom css to a stylesheet.  For the color, you can just add the `g-validation-failed-message` class to the element to keep the style consistent.